ArduPlane/ArduPlane.cpp

```diff
@@ -845,43 +845,117 @@ void Plane::update_flight_stage(void)
   Do we think we are flying?
   Probabilistic method where a bool is low-passed and considered a probability.
 */
-void Plane::determine_is_flying(void)
+void Plane::update_is_flying_5Hz(void)
 {
     float aspeed;
-    bool isFlyingBool;
-
-    bool airspeedMovement = ahrs.airspeed_estimate(&aspeed) && (aspeed >= 5);
+    bool is_flying_bool;
+    uint32_t now_ms = hal.scheduler->millis();
+
+    float ground_speed_thresh_cm = (g.min_gndspeed_cm > 0) ? labs(g.min_gndspeed_cm*0.9f) : 500;
+    bool gps_confirmed_movement = (gps.status() >= AP_GPS::GPS_OK_FIX_3D) &&
+                                    (gps.ground_speed_cm() >= ground_speed_thresh_cm);
+    bool airspeed_movement = ahrs.airspeed_estimate(&aspeed) && (aspeed >= 7);
+
+    if (now_ms < 10000) {
+        // Don't bother checking in the first 10 seconds of a boot to allow for a GPS init and re-lock.
+        // Note: This time is actually longer on a pixhawk when considering time it takes to check
+        // the Bootloader and NuttX. This may need to be reconsidered for Linux builds.
+        is_flying_bool = false;
+    }
+    else if (ins.is_still()) {
+        // if motionless, we can't possibly be flying. This is a very strict test
+        is_flying_bool = false;
+    }
+    else if(arming.is_armed()) {
+        // when armed assuming flying and we need overwhelming evidence that we ARE NOT flying
+
+        // short drop-outs of GPS are common during flight due to banking which points the antenna in different directions
+        bool gps_lost_recently = (gps.last_fix_time_ms() > 0) && // we have locked to GPS before
+                        (gps.status() < AP_GPS::GPS_OK_FIX_2D) && // and it's lost now
+                        (now_ms - gps.last_fix_time_ms() < 5000); // but it wasn't that long ago (<5s)
+
+        if ((auto_state.last_flying_ms > 0) && gps_lost_recently) {
+            // we've flown before, loosen GPS constraints
+            is_flying_bool = airspeed_movement && // moving through the air
+                    (gps_confirmed_movement || gps_lost_recently); // locked and moving, or recently lost lock
```

Comment:
While combing through this via @WickedShell 's review I noticed that this line gets reduced to is_flying_bool = airspeed_movement && (false || true)
It does what I want it to do: if you recently lost GPS due to a turn or some other transiet then rely on airspeed